modules/godot/databags/scene_tree_databag.cpp

```diff
@@ -54,14 +55,29 @@ Node *SceneTreeDatabag::get_node_or_null_script(const NodePath &p_path) {
 	return get_node_or_null(p_path);
 }
 
+// TOOD: Update these to utilise the type parameter
 Node *SceneTreeDatabag::find_node(const String &p_mask, bool p_recursive, bool p_owner) {
 	// Using root to search the node.
-	return world_ecs->get_tree()->get_root()->find_node(p_mask, p_recursive, p_owner);
+	Node *root = world_ecs->get_tree()->get_root();
+	TypedArray<Node> search = root->find_nodes(p_mask, "", p_recursive, p_owner);
+
+	if (search.size() > 0) {
+		return (Node *)((Object *)search[0]);
```

Comment:
Why double a cast is needed here? Can't you just return the pointer? as it's already `TypedArray<Node>` it should return `Node*`.

Reply:
(Node*)search[0] throws a casting error - it was what I tried first. Seems the compiler doesn't like the explicit Object* operator with a further implicit Node* cast.

Reply:
Ok, not sure why they made like that, can you please use this syntax instead as it's more clean and avoids to use C cast which is less safe than static_cast:
https://github.com/godotengine/godot/blob/ca101fe080af6012abde372733e5f02a6f67860e/editor/import/scene_import_settings.cpp#L383

```
Node *collider_view = static_cast<Node*>(descendants[0].operator Object *());
```